stglib/aqd/aqdutils.py

```diff
@@ -1420,3 +1435,30 @@ def apply_wave_coord_output(ds, T, T_orig):
         ds = ds.drop(["VEL1", "VEL2", "VEL3"])
 
     return ds
+
+
+def fill_agc(ds):
+    """
+    Fill velocity data with AGC threshold.
+    Beam AGC variables (AGC1_1221, AGC2_1222, and AGC3_1223) are used to fill corresponding beam velocities (vel1_1277, vel2_1278, vel3_1279).
```

Comment:
From what I can tell, the code does not do anything with the beam velocity data.